File: src/lib/validations/cliente.js
import { z } from 'zod';

export const clienteSchema = z.object({
	nombre: z.string().min(1, 'El nombre es requerido'),
	empresa: z.string().optional().or(z.literal('')),
	rfc: z.string().optional().or(z.literal('')),
	correo: z.string().min(1, 'El correo es requerido').email('El correo no es válido'),
	telefono: z.string().optional().or(z.literal('')),
	direccion: z.string().optional().or(z.literal('')),
	notas: z.string().optional().or(z.literal(''))
});

File: mobile/lib/models/cliente.dart
class Cliente {
  final String id;
  final String nombre;
  final String? empresa;
  final String? rfc;
  final String correo;
  final String? telefono;
  final String? direccion;
  final String? notas;
  final bool activo;

  Cliente({
    required this.id,
    required this.nombre,
    this.empresa,
    this.rfc,
    required this.correo,
    this.telefono,
    this.direccion,
    this.notas,
    this.activo = true,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) => Cliente(
        id: json['id'] ?? '',
        nombre: json['nombre'] ?? '',
        empresa: json['empresa'],
        rfc: json['rfc'],
        correo: json['correo'] ?? '',
        telefono: json['telefono'],
        direccion: json['direccion'],
        notas: json['notas'],
        activo: json['activo'] ?? true,
      );
}
